misk/src/main/kotlin/misk/jdbc/DataSourceModule.kt

```diff
@@ -4,47 +4,55 @@ import com.google.inject.Key
 import com.zaxxer.hikari.HikariConfig
 import com.zaxxer.hikari.HikariDataSource
 import misk.inject.KAbstractModule
-import misk.inject.asSingleton
 import javax.inject.Inject
 import javax.inject.Provider
-import javax.sql.DataSource
 import kotlin.reflect.KClass
 
-/** Binds a [DataSource] configured from the specific entry in the [DataSourcesConfig] */
+/**
+ * Binds a [DataSourceCluster] configured from the specific entry in the [DataSourceClustersConfig]
+ */
 class DataSourceModule constructor(
   private val datasourceName: String,
-  private val key: Key<DataSource>
+  private val key: Key<DataSourceCluster>
 ) : KAbstractModule() {
 
   override fun configure() {
-    bind(key).toProvider(DataSourceProvider(datasourceName)).asEagerSingleton()
+    bind(key).toProvider(DataSourceClusterProvider(datasourceName)).asEagerSingleton()
   }
 
   companion object {
     fun create(datasourceName: String) =
-        DataSourceModule(datasourceName, Key.get(DataSource::class.java))
+        DataSourceModule(datasourceName, Key.get(DataSourceCluster::class.java))
 
     fun create(datasourceName: String, annotatedBy: Annotation) =
-        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy))
+        DataSourceModule(datasourceName, Key.get(DataSourceCluster::class.java, annotatedBy))
 
     fun <A : Annotation> create(datasourceName: String, annotatedBy: Class<A>) =
-        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy))
+        DataSourceModule(datasourceName, Key.get(DataSourceCluster::class.java, annotatedBy))
 
     fun <A : Annotation> create(datasourceName: String, annotatedBy: KClass<A>) =
-        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy.java))
+        DataSourceModule(datasourceName, Key.get(DataSourceCluster::class.java, annotatedBy.java))
   }
 
-  private class DataSourceProvider(private val datasourceName: String) : Provider<DataSource> {
-    @Inject lateinit var datasourcesConfig: DataSourcesConfig
+  private class DataSourceClusterProvider(
```

Comment:
do you still think we should make it a common practice to bind `T` to the `writer`? for example `@Inject DataSource theWriter`

Reply:
I was thinking about this too. I like this idea. In particular when you don't provide specific `reader` configuration and only care about a single rw database instance.

Reply:
I think I misread your comment yesterday so I want to confirm: are you suggesting the `DataSourceModule` should _also_ bind `T` to the `writer` so that application developers don't need to it themselves?

Reply:
yes. basically default to the writer